packages/_private/vue-demo/src/DemoForm.vue

```diff
@@ -0,0 +1,339 @@
+<script setup lang="ts">
+import { ref } from 'vue';
+import type { SynChangeEvent } from '@synergy-design-system/components';
+import {
+  SynVueButton,
+  SynVueCheckbox,
+  SynVueDivider,
+  SynVueInput,
+  SynVueOptgroup,
+  SynVueOption,
+  SynVueRadio,
+  SynVueRadioGroup,
+  SynVueSelect,
+  SynVueSwitch,
+  SynVueTextarea,
+} from '@synergy-design-system/vue';
+import DemoFieldset from './DemoFieldset.vue';
+import { normalizeData } from './shared';
+
+type FormEnabledElements = HTMLElement & {
+  checked?: boolean;
+  name: string;
+  value?: string;
+};
+
+const initialFormData = {
+  code: '',
+  comment: '',
+  date: '',
+  email: '',
+  gender: '',
+  name: '',
+  newsletterAngular: false,
+  newsletterBeta: false,
+  newsletterReact: false,
+  newsletterStandard: false,
+  newsletterVanilla: false,
+  newsletterVue: true,
+  password: 'abAB1234',
+  phone: '',
+  role: '',
+  topics: [],
+};
+
+const formData = ref({
+  ...initialFormData,
+});
+
+const formRef = ref<HTMLFormElement>();
+
+const reset = () => {
+  console.log('here')
+  console.log(formData.value)
+  formData.value = {
+    ...initialFormData,
+  };
+  console.log(formData.value)
+}
+
+const submit = (e: Event) => {
+  e.preventDefault();
+  e.stopPropagation();
+
+  const formElement = e.target as HTMLFormElement;
+  const isValid = formElement.checkValidity();
+
+  if (isValid) {
+    // eslint-disable-next-line no-alert
+    alert('Your data was successfully submitted');
+  }
```

Comment:
We should make the validation handling the same for each wrapper. in *angular* and here in *vue* we only do an alert if its valid, in *vanilla* and *react* we do following:
![image](https://github.com/synergy-design-system/synergy-design-system/assets/115139104/df294503-71d6-47e0-aa52-c5dfd0364c9b)